src/hotspot/cpu/x86/macroAssembler_x86.hpp

```diff
@@ -189,10 +194,17 @@ class MacroAssembler: public Assembler {
   }
   void movdbl(Address dst, XMMRegister src) { movsd(dst, src); }
 
-  void incrementl(AddressLiteral dst, Register rscratch = noreg);
-  void incrementl(ArrayAddress   dst, Register rscratch);
+  void flt_to_flt16(Register dst, XMMRegister src, XMMRegister tmp) {
+    // Instruction requires different XMM registers
+    vcvtps2ph(tmp, src, 0x04, Assembler::AVX_128bit);
```

Comment:
vcvtps2ph can have source and destination as same. Did you mean to say here in the comment that "Instruction requires XMM register as destination"?